src/domain/datasets/services/src/services/key_blocks/dataset_key_block_indexing_job.rs

```diff
@@ -166,6 +166,7 @@ pub(crate) async fn collect_dataset_key_blocks_in_range(
     // Resulting blocks
     let mut key_blocks = Vec::new();
 
+    // TODO: PERF: iterate only over key blocks from (!) storage
```

Comment:
I started implementation, even created `SearchKeyBlocksVisitor`, but got stuck debugging situations where `MetadataChainDatabaseBackedImpl` needs to call iteration, but without cache.

In order not to delay the resolution of the current situation, I suggest we return to this later.